fmn/rules/utils.py

```diff
@@ -79,18 +79,91 @@ def get_packagers_of_package(config, package):
         _cache.configure(**config['fmn.rules.cache'].copy())
 
     key = cache_key_generator(get_packagers_of_package, package)
-    creator = lambda: _get_pkgdb2_packagers_for(config, package)
+    creator = lambda: _get_packagers_for(config, package)
     return _cache.get_or_create(key, creator)
 
 
+def _get_packagers_for(config, package):
+    """ Get the packagers (users) associated with a given package.
+
+    The list is gathered from either pkgdb2 (old) or pagure (new) depending on
+    the configuration value of ``fmn.rules.utils.use_pagure_for_ownership``.
+
+    Args:
+        config (dict): The application configuration.
+        package (str): The package name.
+    Returns:
+        set:  A `set` of packager usernames.
+    """
+    if config['fmn.rules.utils.use_pagure_for_ownership']:
+        return _get_pagure_packagers_for(config, package)
+    else:
+        return _get_pkgdb2_packagers_for(config, package)
+
+
+def _get_pagure_packagers_for(config, package):
+    """ Get the packagers (users) associated with a given package in pagure.
+
+    Args:
+        config (dict): The application configuration.
+        package (str): The package name.
+    Returns:
+        set:  A `set` of packager usernames.
+    """
+    log.debug("Requesting pagure packagers of package %r" % package)
+    base = config.get('fmn.rules.utils.pagure_api_url',
+                      'https://src.fedoraproject.org/pagure/api')
+
+    # XXX. give a default namespace if one is not provided
+    # https://github.com/fedora-infra/fmn/issues/208
+    if '/' not in package:
+        package = 'rpms/' + package
+
+    url = '{0}/0/{1}'.format(base, package)
+    log.info("Querying Pagure at %s for packager information", url)
+    try:
+        response = requests.get(url, timeout=15)
+    except requests.exceptions.ConnectTimeout as e:
+        log.warn('URL %s timed out %r', response.url, e)
+        return set()
+
+    if response.status_code != 200:
+        log.warn('URL %s returned code %s', response.url, response.status_code)
+        return set()
+
+    data = response.json()
+
+    packagers = set(sum(data['access_users'].values(), []))
```

Comment:
Nope - them too.  This is for routing notifications to packagers about packages "they are interested in"... and the old pkgdb logic here included everyone with any kind of association with the package.